src/main/resources/+ciplugins/+jenkins/TestResultsViewPlugin.m

```diff
@@ -0,0 +1,33 @@
+classdef TestResultsViewPlugin < matlab.unittest.plugins.TestRunnerPlugin
+    % Copyright 2025 The MathWorks, Inc.
+    
+    methods (Access=protected)
+        function reportFinalizedSuite(plugin, pluginData)
+            % Checkout MATLAB Test license
+            license('checkout', 'matlab_test');
+
+            testDetails = {};
+            for idx = 1:numel(pluginData.TestResult)
+                testDetails(idx).TestResult = pluginData.TestResult(idx);
+                testDetails(idx).BaseFolder = pluginData.TestSuite(idx).BaseFolder;
+            end
+
+            % If test results artifact exists, update the same file
+            testArtifactFile = fullfile(getenv("MW_MATLAB_TEMP_FOLDER"),"matlabTestResults.json");
+            if isfile(testArtifactFile)
+                testResults = {jsondecode(fileread(testArtifactFile))};
+            else
+                testResults = {};
+            end
+            testResults{end+1} = testDetails;
+            JsonTestResults = jsonencode(testResults, "PrettyPrint", true);
+            
+            fID = fopen(testArtifactFile, "w");
+            fprintf(fID, '%s', JsonTestResults);
+            fclose(fID);

```

Comment:
Check how we handle writing to the file in the `BuildReportPlugin` class. We issue a warning if `fopen` returns `-1`, and clean up a little differently using `onCleanup` (which is safer). I think this was the source of a bug when we first introduced the build report to the Jenkins plugin. Otherwise it might fail the test session/ build because of a weird filesystem permissions issue.

Reply:
Updated, thanks!